onnxruntime/core/providers/dml/dml_provider_factory.cc

```diff
@@ -216,6 +254,107 @@
   API_IMPL_END
 }
 
+bool IsHardwareAdapter(IDXCoreAdapter* adapter) {
+    bool is_hardware{ false };
```

Comment:
>{ false }; [](http://example.com/codeflow?start=20&length=10)

[nit] There's an odd mix of normal assignment and in-place initialization. e.g.

`OrtDmlDeviceFilter dev_filter = device_opts->dev_filter;`
`bool is_hardware{ false };`

It would be nice to be consistent. Except for actual constructor calls, ordinary assignment is slightly cleaner to read (fewer characters to type, visually stands out more, and more consistent with existing precedent).

`bool is_hardware = false;`